Expose nholmann json to torch

c-p-i-o opened the pull request:
Summary:



Expose nlohmann json library so that it can be used from inside Pytorch. The library already exists in the `third_party` directory. This PR is making `nlohmann/json.hpp` header available to be used from `torch.distributed`.
The next PR makes actual use of this header.

imported-using-ghimport

Test Plan: Imported from OSS

Reviewed By: malfet

Differential Revision: D59035246

Pulled By: c-p-i-o

d4l3k reviewed (approved):
approving since it's same as the other PR

c-p-i-o committed:
Expose nholmann json to torch (#129570) (#129570)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/129570

Expose nlohmann json library so that it can be used from inside Pytorch. The library already exists in the `third_party` directory. This PR is making `nlohmann/json.hpp` header available to be used from `torch.distributed`.
The next PR makes actual use of this header.

imported-using-ghimport

Test Plan: Imported from OSS

Reviewed By: malfet

Differential Revision: D59035246

Pulled By: c-p-i-o

c-p-i-o commented:
Sorry @malfet . Not sure what I did wrong.
I tried landing #129504 but it said D59035246 must be landed first.
When I tried landing D59035246 it said the diff needs to be exported to PyTorch. When I clicked the export button, it made a new PR on github.

malfet reviewed (approved):
Sure, though I wonder why do you need to land it as co-dev, just merge on OSS and internal diff will be landed in the next 24 hours

c-p-i-o commented:
@pytorchbot merge

c-p-i-o commented:
@pytorchbot label "topic: not user facing"

c-p-i-o commented:
@pytorchbot merge

clee2000 commented:
@pytorchbot merge

Cancelled old merge job to fix something, reissuing merge command, sorry about that
